network_simulator/plot_tool.py

```diff
@@ -0,0 +1,300 @@
+        plt.ylabel("Size of Packet Transmitted (Bits)")
+        # Set up grid on graph.
+        plt.grid(True)
+
+    def graph_flows(self):
+        """
+        Graphs the flow stats of each available flow.
+        """
+        # Set up legend storage for flow names.
+        sent_legend = []
+        receive_legend = []
+        rtt_legend = []
+        window_legend = []
+
+        # Iterate through each flow to display each stats.
+        for flow, flow_stats in self.stats.flow_stats.iteritems():
+            # Name of flow would be "flow_id".
+            # Packet sent times.
+            plt.subplot(432)
+            sent_rate_lst = self.tool.gen_rate_tuple_list(
+                flow_stats.packet_sent_times)
+            self.tool.graph_tuple_list(sent_rate_lst)
+            sent_legend.append(flow)
+
+            # Packet receive times.
+            plt.subplot(435)
+            rec_rate_lst = self.tool.gen_rate_tuple_list(
+                flow_stats.packet_rec_times)
+            self.tool.graph_tuple_list(rec_rate_lst)
+            receive_legend.append(flow)
+
+            # Packet RTT times.
+            plt.subplot(438)
+            self.tool.graph_1d_list(flow_stats.packet_rtts)
+            rtt_legend.append(flow)
+
+            # Window size in packets w.r.t time.
+            plt.subplot(4,3,11)
+            self.tool.graph_tuple_list(flow_stats.window_size_times)
+            window_legend.append(flow)
+
+        # Finalize sent graph.
+        plt.subplot(432)
+        plt.legend(sent_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Packet Sent Rate")
+        # Set up grid on graph.
+        plt.grid(True)
+
+        # Finalize receive graph.
+        plt.subplot(435)
+        plt.legend(receive_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Packet Received Rate")
+        # Set up grid on graph.
+        plt.grid(True)
+
+        # Finalize RTT graph.
+        plt.subplot(438)
+        plt.legend(rtt_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("RTT (sec)")
+        # Set up grid on graph.
+        plt.grid(True)
+
+        # Finalize window size graph.
+        plt.subplot(4,3,11)
+        plt.legend(window_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Window Size (pkts)")
+        # Set up grid on graph.
+        plt.grid(True)
+
+    def graph_hosts(self):
+        """
+        Graphs the host stats of each available host.
+        """
+        # Set up legend storage for host names.
+        sent_legend = []
+        receive_legend = []
+
+        # Iterate through each host to display each stats.
+        for host, host_stats in self.stats.host_stats.iteritems():
+            # Name of host would be "host_address".
+            # Packet sent times.
+            plt.subplot(433)
+            sent_rate_lst = self.tool.gen_rate_tuple_list(
```

Comment:
Divide all rates' y-axes by MEGABIT as mentioned before (this should be done for all rates in other functions too). Something like sent_rate_lst = [(a[0], float(a[1]) / MEGABIT) for a in sent_rate_lst] should work I think.